### PR TITLE
Ch11

### PR DESCRIPTION
**Pull request by joelbirchler**
A book library application that connects to Postgres

### CODE REVIEW

File: books/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+WORKDIR $HOME/go/src/books
+COPY *.go .
+
+RUN go get -u \
+      github.com/gorilla/mux \
+      github.com/lib/pq && \
+    go build
+
+
+# -- App stage --
```
Comment (imcsk8) [inline books/Dockerfile:14]: A multi stage Dockerfile seems a little much for this application. 
I would suggest to add OpenShift labels, with them OpenShift will create the service for the application for free, check this Dockerfile for reference: https://github.com/RedHatTraining/DO280-apps/blob/master/famous-quotes/Dockerfile#L8

File: books/Makefile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+all: fmt build run
+
+fmt:
+	gofmt -w *.go
+
+build:
+	sudo buildah bud -t books:latest .
+
+run:
+	sudo podman run \
```
Comment (imcsk8) [inline books/Makefile:13]: Why sudo?
Lates podman should work when run as normal user.

**Review by imcsk8 — CHANGES_REQUESTED** [review]
The Makefile should work without sudo

File: books/Dockerfile (hunk 1 above)
Comment (joelbirchler) [reply books/Dockerfile:14]: I think labels in images breaks the responsibility lines / separation of concerns / reusability.

It looks like OpenShift team might agree with me. They appear to be removed in OpenShift 4? I couldn't find any documentation and then I found this: https://bugzilla.redhat.com/show_bug.cgi?id=1583274

File: books/Makefile (hunk 2 above)
Comment (joelbirchler) [reply books/Makefile:13]: I switched it to sudo when buildah was having networking issues (egress), but will go back and troubleshoot. I believe there was a bug that is patched in the latest versions of buildah, but not available on RHEL yet. :/

**Comment by joelbirchler** [comment]
I'm making some bigger changes here in order to output a different image that will appear to be a second app. Closing this PR to avoid merge conflicts.

**Comment by joelbirchler** [comment]
Reopened this. It now have two applications that are very similar. Both are used in Chapter 11. I made some attempts to create a single code base that would abstract away the duplication, but the complexity wasn't worth it. It may be worth considering consolidating the source in someway in the future.

File: books/Makefile (hunk 2 above)
Comment (joelbirchler) [reply books/Makefile:13]: It looks like the issue I'm running into is that the distributed version of podman/buildah has a network bug when running rootless: https://github.com/containers/buildah/issues/1223. 

I guess an alternative is to vendor the dependencies.

Comment (ffapitalle) [reply books/Makefile:13]: Changed to work rootless here: bd91e89ab163da3a9e9c23c1b1cd739b872d8b71

**Comment by ffapitalle** [comment]
Merging this PR to make `exoplanets` app source code available. 

- Changed exoplanets Makefile to work without sudo

Remaining issues will be worked in different PRs if needed.

**Pull request merged**